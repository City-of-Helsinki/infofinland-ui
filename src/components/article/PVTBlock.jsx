import Block from '@/components/layout/Block'

export default function PVTBlock({
  field_email_address,
  field_phonenumber,
  field_postal_address,
  field_postal_address_additional,
  field_service_hours,
  field_visiting_address,
  field_visiting_address_additional,
  title,
}) {
  return (
    <Block>
      {title && <h4 className="mb-4 text-h4 font-bold">{title}</h4>}

      {field_visiting_address && (
        <p className="mb-4">{field_visiting_address}</p>
      )}

      {field_visiting_address_additional && (
        <p className="mb-4">{field_visiting_address_additional}</p>
      )}

      {field_postal_address && <p className="mb-4">{field_postal_address}</p>}

      {field_postal_address_additional && (
        <p className="mb-4">{field_postal_address_additional}</p>
      )}

      {field_service_hours && <p className="mb-4">{field_service_hours}</p>}

      {field_phonenumber && <p className="mb-4">{field_phonenumber}</p>}

      {field_email_address && <p className="mb-4">{field_email_address}</p>}
    </Block>
  )
}
