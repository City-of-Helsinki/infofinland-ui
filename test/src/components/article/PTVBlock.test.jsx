/**
 * @jest-environment jsdom
 */

import React from 'react'
// eslint-disable-next-line import/no-extraneous-dependencies
import { render } from '@testing-library/react'
import PVTBlock from '@/components/article/PTVBlock'

const ITEMS = [
  {
    field_email_address: 'aa@bb.com',
    field_phonenumber: ['+358 044 676 9539'],
    field_postal_address: 'Osoite 1 A',
    field_postal_address_additional: 'Kolmas kerros',
    field_service_hours: 'ma-pe: 10-16, la:12-14',
    field_visiting_address: 'Isompi tie 2',
    field_visiting_address_additional: 'Pääovi',
    title: 'Testitiedot',
    id: 'getegeh5g',
  },
  {},
]
describe('Block', () => {
  it('should render the given contact fragments inside a Block', () => {
    const { asFragment } = render(<PVTBlock items={ITEMS} className="tester" />)
    // we can only use toBeInTheDocument because it was imported
    // in the jest.setup.js and configured in jest.config.js
    expect(asFragment()).toMatchSnapshot()
  })
})
