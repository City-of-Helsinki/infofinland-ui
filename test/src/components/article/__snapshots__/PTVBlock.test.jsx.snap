// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Block should render the given contact fragments inside a Block 1`] = `
<DocumentFragment>
  <div
    class="mb-16"
  >
    <div
      class="mb-8 ifu-block--article"
    >
      <h4
        class="mb-4 text-h4 font-bold"
      >
        Testitiedot
      </h4>
      <p
        class=""
      >
        Isompi tie 2
        <br />
        Pääovi
      </p>
      <p
        class="mb-2"
      >
        Osoite 1 A
        <br />
        Kolmas kerros
      </p>
      <p
        class=""
      >
        +358 044 676 9539
      </p>
      <p
        class="mb-4"
      >
        aa@bb.com
      </p>
      <p
        class=""
      >
        ma-pe: 10-16, la:12-14
      </p>
    </div>
    <div
      class="mb-8 ifu-block--article"
    />
  </div>
</DocumentFragment>
`;
